ale/driver_verification.py

```diff
@@ -0,0 +1,228 @@
+    return sun_position
+
+def create_json_dump(driver, sensor_frame_id, target_frame_id):
+    json_dump = {}
+    json_dump["instrument_rotation"] = generate_instrument_rotation(driver, sensor_frame_id)
+    json_dump["body_rotation"] = generate_body_rotation(driver, target_frame_id)
+    json_dump["instrument_position"] = generate_instrument_position(driver)
+    json_dump["sun_position"] = generate_sun_position(driver)
+    return json_dump
+
+def diff_and_describe(json1, json2, key_array):
+    for key in key_array:
+        json1 = json1[key]
+        json2 = json2[key]
+    diff = json1 - json2
+    print(len(diff), np.mean(diff, axis=(0)), np.median(diff, axis=(0)))
+
+
+# Define the function to compare ISDs
+def compare_isds(json1, json2):
+    diff_and_describe(json1, json2, ["instrument_position", "positions"])
+    diff_and_describe(json1, json2, ["instrument_position", "velocities"])
+    diff_and_describe(json1, json2, ["sun_position", "positions"])
+    diff_and_describe(json1, json2, ["sun_position", "velocities"])
+    diff_and_describe(json1, json2, ["instrument_rotation", "quaternions"])
+    diff_and_describe(json1, json2, ["instrument_rotation", "angular_velocities"])
+    diff_and_describe(json1, json2, ["body_rotation", "quaternions"])
+    diff_and_describe(json1, json2, ["body_rotation", "angular_velocities"])
+
+# Define the main function
+def main(driver, image):
+    
+    # Check if the mission name is valid
+    driver_class = driver_exists(driver)
+    if driver_class == None:
+        raise ValueError("Invalid driver name provided.")
+    
+    # Duplicate the image for ALE and ISIS processing
+    image_ale_path = Path(f"{image}_ALE.cub")
+    image_isis_path = Path(f"{image}_ISIS.cub")
+    shutil.copy(image, image_ale_path)
+    shutil.copy(image, image_isis_path)
+        
+    # Run spiceinit with ISIS
+    run_spiceinit_isis(image_isis_path)
+        
+    # Run spiceinit with ALE
+    run_spiceinit_ale(image_ale_path)
+        
+    # Generate ISD for both ALE and ISIS
+    read_ale_driver = ReadIsis(image_ale_path)
+    ale_json_dump = create_json_dump(read_ale_driver, read_ale_driver.sensor_frame_id, read_ale_driver.target_frame_id)
+    read_isis_driver = ReadIsis(image_isis_path)
+    isis_json_dump = create_json_dump(read_isis_driver, read_isis_driver.sensor_frame_id, read_isis_driver.target_frame_id)
+    # because there isn't isis spice drivers for every driver
+    # it is probably better to use tabledump
+        
+    # Compare the ISDs
+    compare_isds(ale_json_dump, isis_json_dump)
+
+    # Clean up duplicated images if needed
+    #image_ale_path.unlink()
+    #image_isis_path.unlink()
+
+# Set up argparse to handle command line arguments
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Script to compare ALE driver and ISIS3 driver against an image.")
+    parser.add_argument('driver', type=str, help='Name of the driver to utilize.')
```

Comment:
Something to note here is that we no longer need to add a driver as an argument. We just use ISIS and hope that ALE has the driver to ingest our image.

Something to add would be to run an ALE load/loads on the image before running spiceinit on it to make sure we have a driver that can make an ISD from the label

Reply:
Can this argument be removed?

Reply:
@acpaquette I finished removing that driver argument. there is a small note to make about the use of pvl.loads() in this script. I commented out 2 lines using that function and had to change to variable names in the subsequent ale.loads() calls because the pvl.loads() function won't work as expected until https://github.com/DOI-USGS/ale/pull/587 is merged to apply the bug fix to that method; that fix will allow ale.loads() to read in PVLObjects, but currently because those lines are commented I changed ale.loads() so that it just reads in the image files themselves. This can be updated to uncomment those lines and use the PVLObjects once that bugfix PR gets merged.

Reply:
I think the change to using the file paths over the PVL objects themselves is fine. Given further thought I don't think we need to be able to read in in-memory PVL objects with the `ale.load|s` calls